src/core/dynamic_batch_scheduler.cc

```diff
@@ -35,18 +35,20 @@
 #include "src/core/logging.h"
 #include "src/core/model_config.h"
 #include "src/core/nvtx.h"
+#include "src/core/server.h"
 
 namespace nvidia { namespace inferenceserver {
 
 DynamicBatchScheduler::DynamicBatchScheduler(
-    const bool dynamic_batching_enabled, const int32_t max_batch_size,
+    TritonModel* model, const bool dynamic_batching_enabled,
```

Comment:
Should use shared_ptr for 'model', not a raw pointer.

Reply:
The scheduler object is held within TritonModel(actually InferenceBackend from which TritonModel extends from). Currently we can not expose the TritonModel class to backend.h as it interferes with ensemble_scheduler. The raw pointer to model should be safe inside Scheduler as the scheduler is owned by TritonModel object itself. I understand the need to avoid raw pointers. But I think we can address it as a separate clean up effort to remove InferenceBackend. I will document the requirement.
Forcing a use of shared_ptr will destroy backend alongwith scheduler.